modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java

```diff
@@ -0,0 +1,230 @@
+        assertThat(tableImpl().internalTable().partitions(), greaterThan(1));
+
+        int oldLatestCatalogVersion = latestCatalogVersion();
+
+        runInTx(commit, tx -> {
+            long[] beforeInserts = partitionSizes();
+
+            int id = 0;
+
+            do {
+                insertPeople(tx, TABLE_NAME, new Person(id, String.valueOf(id), id + 0.0));
+
+                id++;
+            } while (differences(beforeInserts, partitionSizes()) < 2);
+
+            fakeUpdateCatalog();
+
+            assertTrue(isNodeFinishedRwTransactionsStartedBeforeFromNetwork(oldLatestCatalogVersion));
+            assertFalse(isNodeFinishedRwTransactionsStartedBeforeFromNetwork(latestCatalogVersion()));
+        });
+
+        assertTrue(isNodeFinishedRwTransactionsStartedBeforeFromNetwork(oldLatestCatalogVersion));
+        assertTrue(isNodeFinishedRwTransactionsStartedBeforeFromNetwork(latestCatalogVersion()));
+    }
+
+    private static void runInTx(boolean commit, Consumer<Transaction> consumer) {
+        Transaction tx = node().transactions().begin(new TransactionOptions().readOnly(false));
+
+        try {
+            consumer.accept(tx);
+        } finally {
+            assertThat(commit ? tx.commitAsync() : tx.rollbackAsync(), willCompleteSuccessfully());
+        }
+    }
+
+    private static TableImpl tableImpl() {
+        CompletableFuture<Table> tableFuture = node().tables().tableAsync(TABLE_NAME);
+
+        assertThat(tableFuture, willBe(notNullValue()));
+
+        return (TableImpl) tableFuture.join();
+    }
+
+    private static long[] partitionSizes() {
+        InternalTable table = tableImpl().internalTable();
+
+        return IntStream.range(0, table.partitions())
+                .mapToLong(partitionId -> table.storage().getMvPartition(partitionId).rowsCount())
+                .toArray();
+    }
+
+    private static int differences(long[] partitionSizes0, long[] partitionsSizes1) {
+        assertEquals(partitionSizes0.length, partitionsSizes1.length);
+
+        return (int) IntStream.range(0, partitionSizes0.length)
+                .filter(i -> partitionSizes0[i] != partitionsSizes1[i])
+                .count();
+    }
+
+    private static void fakeUpdateCatalog() {
+        int oldLatestCatalogVersion = latestCatalogVersion();
+
+        String oldZoneName = zoneNameForUpdateCatalogVersionOnly;
+        String newZoneName = zoneNameForUpdateCatalogVersionOnly + 0;
+
+        sql(String.format("ALTER ZONE %s RENAME TO %s", oldZoneName, newZoneName));
+
+        zoneNameForUpdateCatalogVersionOnly = newZoneName;
+
+        assertThat(latestCatalogVersion(), greaterThan(oldLatestCatalogVersion));
+    }
+
+    private static int latestCatalogVersion() {
+        return node().catalogManager().latestCatalogVersion();
+    }
+
+    private static IgniteImpl node() {
+        return CLUSTER.node(0);
+    }
+
+    private static boolean isNodeFinishedRwTransactionsStartedBeforeFromNetwork(int catalogVersion) {
```

Comment:
Not sure. Do you have tests, that verify that in case of 1PC transaction we will increment and decrement the counter.
I mean simple KV.put(null, 1, "val1");  It's important to verify that increment and decrement were called.

Reply:
Added a test to check 1PC `org.apache.ignite.internal.index.ItIndexNodeFinishedRwTransactionsCheckerTest#testOnePhaseCommitViaKeyValue`

Reply:
It's not 1PC. Please do not create an explicit transaction, let it be an implicit one. I mean, literally do KV.put() without any runInTx() or similar.

Reply:
Fix it